projects/gloo/cli/pkg/cmd/gatewayapi/convert/domain.go

```diff
@@ -133,7 +153,153 @@ func (o *GatewayAPIOutput) AddErrorFromWrapper(errType ErrorType, wrapper domain
 
 func NewGatewayAPIOutput() *GatewayAPIOutput {
 	return &GatewayAPIOutput{
-		gatewayAPICache: &domain.GatewayAPICache{},
-		edgeCache:       &domain.GlooEdgeCache{},
+		gatewayAPICache: &GatewayAPICache{},
+		edgeCache:       &snapshot.Instance{},
+	}
+}
+
+type GatewayAPICache struct {
+	YamlObjects         []*snapshot.YAMLWrapper
+	HTTPRoutes          map[string]*snapshot.HTTPRouteWrapper
+	RouteOptions        map[string]*snapshot.RouteOptionWrapper
+	VirtualHostOptions  map[string]*snapshot.VirtualHostOptionWrapper
+	ListenerOptions     map[string]*snapshot.ListenerOptionWrapper
+	HTTPListenerOptions map[string]*snapshot.HTTPListenerOptionWrapper
+	DirectResponses     map[string]*snapshot.DirectResponseWrapper
+
+	Upstreams    map[string]*snapshot.UpstreamWrapper
+	AuthConfigs  map[string]*snapshot.AuthConfigWrapper
```

Comment:
Do we run the risk of double-counting? We have some resources that are used in the GatewayAPICache and also used in our EdgeCache

Reply:
i needed both caches to have these configs because not all are passed over. Upstreams for instance will only transfer if they arent kube upstreams.